### PR TITLE
Fix/minor compilation warning extinction.

### PR DESCRIPTION
**Pull request by fabianbl**
ptal @helenol @mfehr

### CODE REVIEW

**Comment by helenol** [comment]
LGTM thanks

File: voxblox/include/voxblox/core/block_inl.h [hunk 1]
```diff
@@ -46,9 +46,8 @@ void Block<VoxelType>::getProto(BlockProto* proto) const {
 }
 
 template <typename VoxelType>
-bool Block<VoxelType>::mergeBlock(const Block<VoxelType>& other_block) {
-  // TODO(mfehr): implement
-  LOG(FATAL) << "NOT IMPLEMENTED";
```
Comment (mfehr) [inline voxblox/include/voxblox/core/block_inl.h]: Thanks for fixing the warning, but why remove this?

File: voxblox/include/voxblox/core/layer_inl.h [hunk 2]
```diff
@@ -165,7 +165,8 @@ bool Layer<VoxelType>::addBlockFromProto(const BlockProto& block_proto,
         if (it == block_map_.end()) {
           block_map_[block_index] = block_ptr;
         } else {
-          it->second->mergeBlock(*block_ptr);
+          CHECK(it->second->mergeBlock(*block_ptr)) << "Block merging not "
```
Comment (mfehr) [inline voxblox/include/voxblox/core/layer_inl.h]: I rather have the crash coming from the function that is not implemented yet. Using the return type as a indicator whether or not the function is implemented is wrong, for some voxel type the return type might be used to communicate some informatio that you can't or shouldn't merge the two blocks.

Comment (fabianbl) [inline voxblox/include/voxblox/core/layer_inl.h]: @mfehr Good point, I changed it back.

**Comment by mfehr** [comment]
@fabianbl thx for fixing this. lgtm

**Pull request merged**